service/module.go

```diff
@@ -23,10 +23,19 @@ package service
 import (
 	"fmt"
 	"sync"
-
-	"github.com/uber-go/tally"
 )
 
+// ModuleProvider provides Modules.
+type ModuleProvider interface {
+	// Name is the name that will be used for a new Module
+	// if no name is given as a ModuleOption.
+	Name() string
```

Comment:
I still think `DefaultName` describes it better and needs less comments to explain what it does.